File: Data Structures & Algorithms/three-integer-sum/submission-10.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {

        Set<List<Integer>> tempSet = new HashSet<>();

        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                for (int k = j + 1; k < nums.length; k++) {
                    if (nums[i] + nums[j] + nums[k] == 0) {
                        List<Integer> subResult = new ArrayList<>(Arrays.asList(nums[i], nums[j], nums[k]));
                        Collections.sort(subResult);

                        tempSet.add(subResult);
                    }
                }
            }

        }
        return new ArrayList<>(tempSet);
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-10.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        int[] prevNums = new int[nums.Length]; 
        int counter = 0;

        for (int i = 0; i < nums.Length; i++) {

            for (int j = 0; j < counter; j++) {
                    if (nums[i] == prevNums[j]) {
                    return true;
                }
            }
            
            prevNums[i] = nums[i];
            counter++;
        }
        return false;
    }
}